bigtable_input.py

```diff
@@ -294,23 +294,24 @@ def moves_from_games(self, start_game, end_game, moves, shuffle,
         # of the total moves, then trimming the result.
         total_moves = self.count_moves_in_game_range(start_game, end_game)
         probability = moves / (total_moves * 0.99)
-        utils.dbg('Row range: %s - %s; total moves: %d; probability %.3f' % (
-            start_row, end_row, total_moves, probability))
+        utils.dbg('Row range: %s - %s; total moves: %d; probability %.3f; moves %d' % (
+            start_row, end_row, total_moves, probability, moves))
         shards = 8
         ds = self.tf_table.parallel_scan_range(start_row, end_row,
                                                probability=probability,
-                                               num_parallel_scans=shards,
                                                columns=[(column_family, column)])
+        ds = ds.take(moves)
         if shuffle:
+            buffer_size = 128
             rds = tf.data.Dataset.from_tensor_slices(
                 tf.random_shuffle(tf.range(0, shards, dtype=tf.int64)))
             ds = rds.apply(
                 tf.contrib.data.parallel_interleave(
                     lambda x: ds.shard(shards, x),
-                    cycle_length=shards, block_length=1024,
+                    cycle_length=shards,
+                    buffer_output_elements=buffer_size,
                     sloppy=True))
-            ds = ds.shuffle(shards * 1024 * 2)
-        ds = ds.take(moves)
+            ds = ds.shuffle(shards * buffer_size * 2)
```

Comment:
this is a smaller shuffle, right? (shards * 128 * 2 vs shards * 1024 * 2).  Can we keep the larger one or is this important to the fix?

Reply:
Yes, I think this final shuffer buffer can remain at its original size, or possibly a little larger, since it's the final shuffle. I'll test the performance effect of some larger sizes.